tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp

```diff
@@ -755,7 +755,7 @@ static bool passCursorInfoForDecl(const ValueDecl *VD,
         PO.SkipIntroducerKeywords = true;
         PO.ArgAndParamPrinting = PrintOptions::ArgAndParamPrintingMode::ArgumentOnly;
         XMLEscapingPrinter Printer(RelDeclsStream);
-        if (BaseType) {
+        if (BaseType && RelatedDecl->getDeclContext() == BaseType->getAnyNominal()) {
```

Comment:
This is a bit sloppy. Where is RelatedDecl coming from and why is it out of sync with BaseType here? There was another hack fix recently in this area to just disable cursor info for generics. Please try to understand the root cause of such issues and refactor the code appropriately instead of just disabling bits of code where you hit assertions.